gst/nnstreamer/tensor_if/gsttensorif.c

```diff
@@ -942,6 +969,57 @@ gst_tensor_if_get_comparison_result (GstTensorIf * tensor_if,
   return TRUE;
 }
 
+#define get_double_val(type, in, out) do { \
+    switch (type) { \
+      case _NNS_INT32: out = (double) (*(int32_t *) in); break; \
+      case _NNS_UINT32: out = (double) (*(uint32_t *) in); break; \
+      case _NNS_INT16: out = (double) (*(int16_t *) in);break; \
+      case _NNS_UINT16: out = (double) (*(uint16_t *) in); break; \
+      case _NNS_INT8: out = (double) (*(int8_t *) in); break; \
+      case _NNS_UINT8: out = (double) (*(uint8_t *) in); break; \
+      case _NNS_FLOAT64: out = (double) (*(double *) in); break; \
+      case _NNS_FLOAT32: out = (double) (*(float *) in); break; \
+      case _NNS_INT64: out = (double) (*(int64_t *) in); break; \
+      case _NNS_UINT64: out = (double) (*(uint64_t *) in); break; \
+      default: g_assert (0); break; \
+    } \
+  } while (0)
+
+/**
+ * @brief Calculate average value of the nth tensor
+ */
+static void
+gst_tensor_if_get_tensor_average (GstTensorIf * tensor_if,
+    GstBuffer * buf, tensor_if_data_s * cv, gint nth)
+{
+  GstMemory *in_mem;
+  GstMapInfo in_info;
+  uint32_t i, size, dsize;
+  const uint32_t *in_dim;
+  double avg = 0.0;
+  double val;
+
+  tensor_type type = tensor_if->in_config.info.info[nth].type;
+  dsize = gst_tensor_get_element_size (type);
+
+  in_dim = tensor_if->in_config.info.info[nth].dimension;
+  size = in_dim[0] * in_dim[1] * in_dim[2] * in_dim[3];
+
+  in_mem = gst_buffer_peek_memory (buf, nth);
+  gst_memory_map (in_mem, &in_info, GST_MAP_READ);
+
+  for (i = 0; i < size; i++) {
+    get_double_val (type, &in_info.data[i * dsize], val);
+    avg = (val - avg) / (i + 1) + avg;
```

Comment:
How about just accumulating `val`, and divide it by size?

Reply:
To prevent overflow.
In the case of uint8, overflow will occur easily.